cloudferrylib/os/network/neutron.py

```diff
@@ -818,31 +818,74 @@ def upload_sec_group_rules(self, sec_groups):
     def upload_networks(self, networks):
         existing_nets_hashlist = (
             [ex_net['res_hash'] for ex_net in self.get_networks()])
+
+        # we need to handle duplicates in segmentation ids
+        used_seg_ids = [i.get('provider:segmentation_id') for i in networks]
+
+        def get_free_segmentation_id():
+            # this function finds first free seg_id
+            # adds it to array of used ids
+            # and returns seg_id
+            for i in xrange(1, 4096):
```

Comment:
This will only work for VLANs and only if `network_vlan_ranges` neutron config option is set to 1:4096. Ranges for other segmentation types are different:

 * VXLANs
 ** (24-bit segmentation ID)[http://tools.ietf.org/html/rfc7348#section-5], meaning ranges from 0 to (2^24)-1;
 ** `vni_ranges` config option in neutron specifies the actual range used in cloud
 * GREs do not require segmentation IDs to be identical on source and destination since it's a tunneling protocol, it only requires the numbers to be set for a particular tenant.

With that in mind, I guess the correct way would be:
```
if segmentation_type in [ "VLAN", "VXLAN" ]:
    segmentation_id = <SRC_SEGMENTATION_ID>
    for _ in number_of_retries:
        try:
            neutron net-create --segmentation-id= <netname>
            break
        except SegmentationIdConflict:
            segmentation_id += 1
            continue
elif segmentation_type == "GRE":
    neutron net-create <netname>
```